tests/conftest.py

```diff
@@ -85,47 +85,59 @@ def distmap(boxarr):
 
 
 @pytest.fixture(scope="function", params=list(itertools.product([1, 3], [0, 1])))
-def make_mfab(boxarr, distmap, request):
+def mfab(boxarr, distmap, request):
     """MultiFab that is either managed or device:
     The MultiFab object itself is not a fixture because we want to avoid caching
     it between amr.initialize/finalize calls of various tests.
     https://github.com/pytest-dev/pytest/discussions/10387
     https://github.com/pytest-dev/pytest/issues/5642#issuecomment-1279612764
     """
 
-    def create():
-        num_components = request.param[0]
-        num_ghost = request.param[1]
-        mfab = amr.MultiFab(boxarr, distmap, num_components, num_ghost)
-        mfab.set_val(0.0, 0, num_components)
-        return mfab
+    class MfabContextManager:
+        def __enter__(self):
+            num_components = request.param[0]
+            num_ghost = request.param[1]
+            self.mfab = amr.MultiFab(boxarr, distmap, num_components, num_ghost)
```

Comment:
`pylint` would probably create warnings for this.

Reply:
Oh really? Linting still coming in #208 😅

CodeQL did not yet complain, what's to be avoided here?

Reply:
Pylint will complain about attributes being declared/created in a non-init function.